src/ServiceControl.AcceptanceTesting/EndpointConfigurationExtensions.cs

```diff
@@ -1,33 +1,96 @@
+    using NServiceBus.AcceptanceTesting.Support;
     using NServiceBus.Configuration.AdvancedExtensibility;
     using NServiceBus.Features;
+    using NServiceBus.Hosting.Helpers;
 
     public static class EndpointConfigurationExtensions
     {
-        public static void NoImmediateRetries(this EndpointConfiguration configuration)
+        // This logic has been adapted from Cores ScanTypesForTest to specialize it to the needs of ServiceControl
+        public static void ScanTypesForTest(this EndpointConfiguration config,
+            EndpointCustomizationConfiguration customizationConfiguration)
         {
-            configuration.Recoverability().Immediate(x => x.NumberOfRetries(0));
+            // disable file system scanning for better performance
+            // note that this might cause issues when required assemblies are only being loaded at endpoint startup time
+            var assemblyScanner = new AssemblyScanner { ScanFileSystemAssemblies = false };
+
+            config.TypesToIncludeInScan(
+                assemblyScanner.GetScannableAssemblies().Assemblies
+                    .Where(a => !a.FullName!.StartsWith("ServiceControl")) // this prevents handlers, custom checks etc from ServiceControl to be scanned
+                    .Where(a => a != customizationConfiguration.BuilderType.Assembly) // exclude all types from test assembly by default
+                    .SelectMany(a => a.GetTypes())
+                    .Union(GetNestedTypeRecursive(customizationConfiguration.BuilderType.DeclaringType, customizationConfiguration.BuilderType))
+                    .Union(customizationConfiguration.TypesToInclude)
+                    .Except(customizationConfiguration.TypesToExclude)
+                    .ToList());
+            return;
+
+            IEnumerable<Type> GetNestedTypeRecursive(Type rootType, Type builderType)
+            {
+                if (rootType == null)
+                {
+                    throw new InvalidOperationException("Make sure you nest the endpoint infrastructure inside the TestFixture as nested classes");
+                }
+
+                yield return rootType;
+
+                if (typeof(IEndpointConfigurationFactory).IsAssignableFrom(rootType) && rootType != builderType)
+                {
+                    yield break;
+                }
+
+                foreach (var nestedType in rootType.GetNestedTypes(BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic).SelectMany(t => GetNestedTypeRecursive(t, builderType)))
+                {
+                    yield return nestedType;
+                }
+            }
         }
 
-        public static void NoDelayedRetries(this EndpointConfiguration configuration)
+        public static async Task DefinePersistence(this EndpointConfiguration config, RunDescriptor runDescriptor, EndpointCustomizationConfiguration endpointCustomizationConfiguration)
```

Comment:
Should this be named `UseInMemoryPersistence`? Because there is no option to configure a different one, right?

Reply:
I think currently the method is consistently called `DefinePersistence`. I have only moved those a bit. If you don't mind I would rather not do it now.